src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,204 @@
-import { Component } from 'react';
-
-type Props = {};
-type State = {};
-
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
-
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Movie } from '../../types/movie';
+import './NewMovie.scss';
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const defaultMovie: Movie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+  const [movie, setMovie] = useState(defaultMovie);
+  const [validations, setValidation] = useState({
+    title: true,
+    imgUrl: true,
+    imdbUrl: true,
+    imdbId: true,
+  });
+
+  const handeInputChange = (name: string, value: string, required: boolean) => {
+    setMovie(prev => ({ ...prev, [name]: value }));
+
+    if (required) {
+      setValidation(prev => ({ ...prev, [name]: true }));
+    }
+  };
+
+  const handleValidation = (
+    event: React.FocusEvent<HTMLInputElement, Element>,
+  ) => {
+    const { value, required, name } = event.target;
+
+    if (!value && required) {
+      setValidation(prev => ({ ...prev, [name]: false }));
+
+      return;
+    }
+
+    // eslint-disable-next-line max-len
+    const urlRegExp = new RegExp(/^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/);
+
+    if ((name === 'imgUrl' || name === 'imdbUrl') && !urlRegExp.test(value)) {
+      setValidation(prev => ({ ...prev, [name]: false }));
+    }
+  };
+
+  const handeFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(movie);
+    setMovie(defaultMovie);
+  };
+
+  return (
+    <form
+      className="NewMovie"
+      onSubmit={event => handeFormSubmit(event)}
+    >
+      <label className="NewMovie__input">
+        Title
+
+        <input
+          name="title"
+          value={movie.title}
+          type="text"
+          className={classNames(
+            'input',
+            {
+              'is-danger': !validations.title,
+            },
+          )}
+          data-cy="form-title"
+          onChange={
+            ({ target: { name, value, required } }) => (
+              handeInputChange(name, value, required))
+          }
+          onBlur={event => handleValidation(event)}
```

Comment:
you can just pass a function everywhere
```suggestion
          onBlur={handleValidation}
```

Reply:
agree, do it like that everywhere

Reply:
done